### PR TITLE
Upgrade to ruby 2.7.1

### PR DESCRIPTION
**Pull request by cdccollins**
These are the changes necessary for the app to run on Ruby 2.7.1.

I have left the version of Ruby the same, since there is a script that can be run after all the apps have been updated to update the version all at once. 

Still some warnings coming from Rspec-mocks, the fix has not been merged yet

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.6.6
+2.7.0
```
Comment (thomasleese) [inline .ruby-version:1]: Did you mean for this change to be in the PR?

Comment (cdccollins) [reply .ruby-version:1]: Oops, no, changing it now :)

**Comment by kevindew** [comment]
I think it'd be beneficial to include the .ruby-version change as part of this for a few reasons: allows using syntax for 2.7, makes it easier to double check in review the quantity of warnings we'll have to deal with, and one less thing to review/deploy when ready.

For the places where we're doing argument forwarding it'd be good if we could use the `...` operator in 2.7 (sidenote, I'm not sure that operator has a name? I call it the ellipsis one). For example we could change:

https://github.com/alphagov/email-alert-api/blob/8c948193b1c2b07f699664c188b8b723cb7595c5/app/services/application_service.rb#L2-L4

to

```
  def self.call(...)
    new(...).call
  end
```

which would mean we could avoid adding the double splats in: app/services/immediate_email_generation_service.rb, app/services/matched_content_change_generation_service.rb and a few more.

I notice we still get about 5 warnings still about kwargs in 2.7, is that something we're experiencing in most repos? I see CSV and RSpec are the culprits here, do we know if they have upgrade plans?

**Comment by pixeltrix** [comment]
> For the places where we're doing argument forwarding it'd be good if we could use the `...` operator in 2.7 (sidenote, I'm not sure that operator has a name? I call it the ellipsis one). For example we could change:

Operator forwarding was only added in Ruby 2.7 so we'd need to deploy either both the Ruby upgrade and the app at the same time or deploy the Ruby upgrade first and then the app upgrade to fix it with Ruby 2.7. I'd be nervous about using a version specific feature during the deployment process and maybe that's something that should be addressed via a [Rubocop rule][1] in a later deploy?

> I notice we still get about 5 warnings still about kwargs in 2.7, is that something we're experiencing in most repos? I see CSV and RSpec are the culprits here, do we know if they have upgrade plans?

Looks like the CSV error is because the app is using the `CSV()` singleton method which is a pretty dated pattern - we can probably re-write the code to avoid the method. The [RSpec fix] is yet to be merged.

[1]: https://github.com/rubocop-hq/rubocop/pull/7646
[2]: https://github.com/rspec/rspec-mocks/pull/1324

**Comment by pixeltrix** [comment]
@kevindew seems like you suggested to use `CSV()` in this [comment][1] - any preference whether we should replace it with `CSV.instance` or `CSV.new` ?

**Comment by kevindew** [comment]
> @kevindew seems like you suggested to use `CSV()` in this [comment](https://github.com/alphagov/email-alert-api/pull/1364#discussion_r469179526) - any preference whether we should replace it with `CSV.instance` or `CSV.new` ?

Very sorry for the long delay, I started writing a reply but must have never finished.

I just suggested using CSV directly for consistency with [our existing approach](https://github.com/alphagov/email-alert-api/blob/0e04a7537f3302ab010dd13153c19d1529a56068/lib/reports/living_in_europe_report.rb#L70) - I don't know enough about the choice to have a preference, though `.new` sounds more idiomatic. Happy with whatever feels most appropriate.

**Review by benbaumann95 — APPROVED** [review]

**Pull request merged**